framework/db/ActiveQuery.php

```diff
@@ -728,6 +732,78 @@ public function orOnCondition($condition, $params = [])
     }
 
     /**
+     * Transform all relation names in exists/not exists conditions to Query objects.
+     * @param mixed $where
+     * @return array
+     */
+    private function buildExistsRelations($where)
+    {
+        if (!isset($where[0]) || !is_array($where)) {
+            return $where;
+        }
+
+        $operator = array_shift($where);
+        /** @var ActiveRecord $model */
+        $model = new $this->modelClass;
```

Comment:
True. Whats about replace `new $this->modelClass` with `Yii::createObject($this->modelClass)`?